avfilter/hwupload_cuda: add YUVA420P format support
avfilter/overlay_cuda: fix framesync with embedded PGS subtitle
Index: jellyfin-ffmpeg/libavfilter/vf_hwupload_cuda.c
===================================================================
--- jellyfin-ffmpeg.orig/libavfilter/vf_hwupload_cuda.c
+++ jellyfin-ffmpeg/libavfilter/vf_hwupload_cuda.c
@@ -57,7 +57,7 @@ static int cudaupload_query_formats(AVFi
     int ret;
 
     static const enum AVPixelFormat input_pix_fmts[] = {
-        AV_PIX_FMT_NV12, AV_PIX_FMT_YUV420P, AV_PIX_FMT_YUV444P,
+        AV_PIX_FMT_NV12, AV_PIX_FMT_YUV420P, AV_PIX_FMT_YUVA420P, AV_PIX_FMT_YUV444P,
         AV_PIX_FMT_P010, AV_PIX_FMT_P016, AV_PIX_FMT_YUV444P16,
         AV_PIX_FMT_0RGB32, AV_PIX_FMT_0BGR32,
 #if CONFIG_VULKAN
Index: jellyfin-ffmpeg/libavfilter/vf_overlay_cuda.c
===================================================================
--- jellyfin-ffmpeg.orig/libavfilter/vf_overlay_cuda.c
+++ jellyfin-ffmpeg/libavfilter/vf_overlay_cuda.c
@@ -157,9 +157,12 @@ static int overlay_cuda_blend(FFFrameSyn
     if (ret < 0)
         return ret;
 
-    if (!input_main || !input_overlay)
+    if (!input_main)
         return AVERROR_BUG;
 
+    if (!input_overlay)
+        return ff_filter_frame(outlink, input_main);
+
     ret = av_frame_make_writable(input_main);
     if (ret < 0) {
         av_frame_free(&input_main);
